File: optimise/v05.py
import sys
from math import gcd


for line in sys.stdin:
    N = int(line) + 1
    combinations = 0
    for x in range(2, N):
        xx = x * x
        for y in range(x + 1, N):
            yy = y * y
            for z in range(y + 1, N):
                if xx + yy == z * z and gcd(gcd(x, y), z) == 1:
                    combinations += 1
    print(combinations)

File: optimise/v05.c
#include <stdio.h>
#include <numeric>


int main(){
    int N, combinations, xx, yy;
    while (scanf("%d", &N) && N){
        combinations = 0;
        for(int x=2; x < N; x++){
            xx = x*x;
            for(int y=x+1; y < N; y++){
                yy = y*y;
                for(int z=y+1; z < N; z++){
                    if(x < y && y < z &&
                       xx + yy == z * z &&
                       std::gcd(std::gcd(x, y), z) == 1){
                            combinations += 1;
                    }
                }
            }
        }
        printf("%d\n", combinations);
    }
    return 0;
}
